Improve our bundler and gemspec handling.

troessner opened the pull request:
Fixes #821 with some small adjustments:

* I decided to put "development_dependencies" under the group "test" because we use them for running tests local and on travis
* I renamed "local_development" to "development"
* I removed the "yard" gem and the "redcarpet" gem. Why exactly did we have them in the first place? I can't really remember :)

Additionally I bumped `unparser` from 0.2.2 to 0.2.4 since bundler was acting up on ruby 2.1 and 2.0 for whatever reasons. I didn't investigate since this upgrade fixed it and seemed non-critical since we plan to remove `unparser` as soon as possible anyway.

troessner commented:
Update: And I removed "activesupport" since we were not using this anymore ;)

mvz commented:
> I removed the "yard" gem and the "redcarpet" gem. Why exactly did we have them in the first place? 

To create the documentation of course :-).

troessner commented:
But none of us 3 is using this locally, are we? I *never* do that. I am all in for code documentation yarddoc style but i still think we can throw out the gem, cant we?

mvz commented:
I sometimes run it locally to check the result of documentation changes.

mvz commented:
That said, we can probably leave the gem out of the bundle.

mvz commented on Gemfile:
```diff
@@ -3,14 +3,20 @@ source 'https://rubygems.org'
+  gem 'rubocop',       '~> 0.34.0'
+end
 
+group :development do
```
I think I preferred you original development/debugging proposal. With test/development, combined with how the groups are named in Rails, I get confused why rake would be in test.

Oh, maybe the gems in test don't actually need to be in a group?

troessner replied:
I changed it back to my initial suggestion.

troessner commented:
>> That said, we can probably leave the gem out of the bundle.

But you're a team member and should feel comfortable with our gem set up -> I put it back in (and redcarpet ofc).

Addressed all comments! Merge? :)

chastell commented:
> I removed "activesupport" since we were not using this anymore

:tada:

> Merge?

I’m happy to do that. :)

mvz commented:
:+1: thanks, @troessner
